go/test/endtoend/vreplication/vreplication_vtctldclient_cli_test.go

```diff
@@ -184,6 +193,135 @@ func createMoveTables(t *testing.T, sourceKeyspace, targetKeyspace, workflowName
 }
 
+// reshard helpers
+
+func splitShard(t *testing.T, keyspace, workflowName, sourceShards, targetShards string, targetTabs map[string]*cluster.VttabletProcess) {
+	createFlags := []string{"--auto-start=false", "--defer-secondary-keys=false", "--stop-after-copy",
+		"--on-ddl", "STOP", "--tablet-types", "primary,rdonly", "--tablet-types-in-preference-order=true",
+		"--all-cells", "--format=json",
+	}
+	rs := newReshard(vc, &reshardWorkflow{
+		workflowInfo: &workflowInfo{
+			vc:             vc,
+			workflowName:   workflowName,
+			targetKeyspace: keyspace,
+		},
+		sourceShards: sourceShards,
+		targetShards: targetShards,
+		createFlags:  createFlags,
+	}, workflowFlavorVtctld)
+
+	ksWorkflow := fmt.Sprintf("%s.%s", keyspace, workflowName)
+	rs.Create()
+	validateReshardResponse(rs)
+	workflowResponse := getWorkflow(keyspace, workflowName)
+	reshardShowResponse := getReshardShowResponse(&rs)
+	require.EqualValues(t, reshardShowResponse, workflowResponse)
+	validateReshardWorkflow(t, workflowResponse.Workflows)
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	rs.Start()
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	for _, tab := range targetTabs {
+		alias := fmt.Sprintf("zone1-%d", tab.TabletUID)
+		query := "update _vt.vreplication set source := replace(source, 'stop_after_copy:true', 'stop_after_copy:false') where db_name = 'vt_customer' and workflow = '" + workflowName + "'"
+		output, err := vc.VtctlClient.ExecuteCommandWithOutput("ExecuteFetchAsDba", alias, query)
+		require.NoError(t, err, output)
+	}
+	rs.Start()
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Running.String())
+	rs.Stop()
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	rs.Start()
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Running.String())
+	for _, targetTab := range targetTabs {
+		catchup(t, targetTab, workflowName, "Reshard")
+	}
+	vdiff(t, keyspace, workflowName, "zone1", false, true, nil)
+
+	rs.SwitchReadsAndWrites()
+	waitForLowLag(t, keyspace, workflowName+"_reverse")
+	vdiff(t, keyspace, workflowName+"_reverse", "zone1", true, false, nil)
+
+	rs.ReverseReadsAndWrites()
+	waitForLowLag(t, keyspace, workflowName)
+	vdiff(t, keyspace, workflowName, "zone1", false, true, nil)
+	rs.SwitchReadsAndWrites()
+	rs.Complete()
+}
+
+func getReshardShowResponse(rs *iReshard) *vtctldatapb.GetWorkflowsResponse {
+	(*rs).Show()
+	reshardOutput := (*rs).GetLastOutput()
+	var reshardResponse vtctldatapb.GetWorkflowsResponse
+	err := protojson.Unmarshal([]byte(reshardOutput), &reshardResponse)
+	require.NoError(vc.t, err)
+	reshardResponse.Workflows[0].MaxVReplicationTransactionLag = 0
+	reshardResponse.Workflows[0].MaxVReplicationLag = 0
```

Comment:
Another place where I'm not sure why we're doing this so a comment would be nice.